sstables/sstables.cc

```diff
@@ -2616,11 +2624,13 @@ future<bool> validate_checksums(shared_sstable sst, reader_permit permit) {
                 valid = co_await do_validate_compressed<adler32_utils>(data_stream, sst->get_compression(), false, digest);
             }
         } else {
-            auto checksum = co_await sst->read_checksum();
+            co_await sst->read_checksum();
+            const auto checksum = sst->get_checksum();
+            SCYLLA_ASSERT(checksum != nullptr);
```

Comment:
on_internal_error

Reply:
I don't think we should even use `on_internal_error()` here. If the sstable doesn't have the checksum component, we should just `return true`.

Reply:
Would we not just create a regular input_stream then?

Reply:
Running `sstable validate-checksums` from `master` against an uncompressed SSTable without a CRC file fails with a `malformed_sstable_exception`:
```
ERROR 2024-08-21 18:28:51,490 [shard 0:main] sstable - Could not create SSTable component /home/ndragazis/Documents/scylladb/git/scylladb/test/resource/sstables/invalid_boundary/mc-33-big-CRC.db. Found exception: std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed: No such file or directory [/home/ndragazis/Documents/scylladb/git/scylladb/test/resource/sstables/invalid_boundary/mc-33-big-CRC.db])
error running operation: sstables::malformed_sstable_exception (/home/ndragazis/Documents/scylladb/git/scylladb/test/resource/sstables/invalid_boundary/mc-33-big-CRC.db: file not found)
```
So, I guess we should preserve the current behavior and fail immediately.

Reply:
I forgot to consider the case of the missing CRC component when writing `validate_checksum()`. I think we should handle gracefully the missing component. I think `on_internal_error()` is too big of a hammer here, a simple exception is enough.

Reply:
At the same time we need to look at the bigger picture here: what do we want to happen when somebody starts a scrub/validate on all the sstables. An exception can kill the entire scrub process and would be very annoying. We need to make sure the validation runs to completion. So maybe just a warning log is enough.

Reply:
After syncing with @denesb , we decided it's better to do the following:
* uncompressed SSTables with a missing CRC component (file is missing but appears in the TOC): keep the current behavior, that is consider the SSTable as broken and throw a `malformed_sstable_exception`. No changes are needed for this.
* uncompressed SSTables without a CRC component (no entry in the TOC): consider them as valid (i.e., return true) and log a warning. A null pointer returned by `read_checksum()` indicates this case.

Note also that scrub/validate doesn't use `validate_chechsum()` at all.